### PR TITLE
Fix bugs related to year and month edges

### LINKED ISSUE #1: Past date gets formatted as being in the future
I've seen this bug pop up in an app recently:
```
console.log(x)
> Tue Dec 13 2016 18:30:08 GMT+0100 (CET)
console.log(relativeTime.format(x))
> in 11 months
```
I haven't yet had the chance to dig into this repo to look for the cause. Wanted to at least file this bug report though.

### PR DESCRIPTION
**Pull request by jzaefferer**
Fixes #1

### CODE REVIEW

File: src/relative-time.js [hunk 1]
```diff
@@ -59,6 +59,9 @@ export default class RelativeTime {
     // x hours ago or yesterday.
     if (absDiff < day) {
       let diffDays = date.getDate() - now.getDate();
+      if (date.getMonth() !== now.getMonth()) {
+        diffDays += (date.getMonth() - now.getMonth()) * date > now ? daysInMonth(now) : -daysInMonth(date);
+      }
```
Comment (rxaviers) [inline src/relative-time.js]: Do we need year check in addition?

Comment (jzaefferer) [inline src/relative-time.js]: year...

**Comment by jzaefferer** [comment]
Updated. This looks much simpler now and works better.

File: src/relative-time.js [hunk 2]
```diff
@@ -45,6 +45,15 @@ export default class RelativeTime {
     var diff = date.getTime() - now.getTime();
     var absDiff = Math.abs(diff);
     var round = Math[diff > 0 ? "floor" : "ceil"];
+    var diffYears = date.getFullYear() - now.getFullYear();
+    var diffMonths = date.getMonth() - now.getMonth();
+    var diffDays = date.getDate() - now.getDate();
+    if (diffYears) {
+      diffMonths += date > now ? 12 : -12;
```
Comment (rxaviers) [inline src/relative-time.js]: Do we need to add diffDays in here too?

**Comment by rxaviers** [comment]
Thanks @jzaefferer

File: src/relative-time.js (hunk 2 above)
Comment (jzaefferer) [inline src/relative-time.js]: No, I don't think so

**Comment by rxaviers** [comment]
:+1:

**Pull request merged**